submissions/LGeorge007/memory-pair-game/script.js

```diff
@@ -0,0 +1,93 @@
+let initialArray = ["img/cartman.jpg",
+    "img/chief.jpg",
+    "img/sten.jpg",
+    "img/kyle.jpg",
+    "img/kenny.jpg"
+];
+let arrOfUnits;
+const imgCover = `img/cover.jpg`;
+const gameField = document.querySelector("#gameField");
+let isLocked = false;
+let firstCard;
+let secondCard;
+let isSecondCard = false;
+let unopenedCards;
+
+//feed him array & any even number
+const duplicateElements = function(array, times) {
+    return array.reduce((res, current) => {
+        return res.concat(Array(times).fill(current));
+    }, []);
+};
+
+const shuffle = function(arr) {
+    const result = [];
+    while (arr.length > 0) {
+        let random = Math.floor(Math.random() * arr.length);
+        let elem = arr.splice(random, 1)[0];
+        result.push(elem);
+    }
+    return result;
+};
+
+const checkCard = function({ target }) {
+    if (!target.closest(".flipper")) { return };
+    if (!isLocked) {
+        if (isSecondCard) {
+            isLocked = true;
+            secondCard = target.closest(".flipper");
+            secondCard.classList.toggle("rotateCard");
+            //The same card must not be removed
+            if (firstCard.id !== secondCard.id) {
+                if (firstCard.dataset.hero === secondCard.dataset.hero) {
+                    unopenedCards -= 2;
+                    setTimeout(() => {
+                        firstCard.children[1].classList.add("hiddenSide");
+                        secondCard.children[1].classList.add("hiddenSide");
```

Comment:
Don't use constructions like children[1], firstElementChild, parentNode, nextSibling, etc. In such way, you rely on the order of DOM elements. So in case when you will have changed design - your code will be broken. Which is bad, obviously. Use querySelector or closest, if in event, instead.